### ISSUE TITLE
error -11: futurerestore screwed up and forced me to later ver (6+)

### ISSUE THREAD

**Issue by francaisoui** [tihmstar/futurerestore#102]
jacques-iMac:futurerestore-latest jacque$ ./futurerestore_macos -t lol.shsh -b Mav10-5.32.00.Release.bbfw -p BuildManifest.plist -s sep-firmware.n56.RELEASE.im4p -m BuildManifest.plist -w iPhone_5.5_10.2_14C92_Restore.ipsw 
Version: 99da4c33526e5dff9b2841a782a6cf9cc41c0aae - 116
[INFO] 64bit device detected
futurerestore init done
reading ticket lol.shsh done
[TSSC] opening BuildManifest.plist
WARNING: Unable to find BbSkeyId node
[TSSR] User specified not to request a Baseband ticket.
Request URL set to https://gs.apple.com/TSS/controller?action=2
Sending TSS request attempt 1... response successfully received
Did set sep+baseband path and firmware
[TSSC] opening BuildManifest.plist
WARNING: Unable to find BbSkeyId node
[TSSR] User specified to request only a Baseband ticket.
ERROR: Unable to get BasebandFirmware node
ERROR: Unable to find required BbGoldCertId in parameters
Request URL set to https://gs.apple.com/TSS/controller?action=2
Sending TSS request attempt 1... response successfully received
Found device in Normal mode
Entering recovery mode...
INFO: device serial number is (serial)
waiting for nonce: 0e 53 f3 3e 3f 37 04 17 43 97 e7 2e 43 93 01 fe 94 19 63 c0 
Got ApNonce from device: 0e 53 f3 3e 3f 37 04 17 43 97 e7 2e 43 93 01 fe 94 19 63 c0 
Device has requested ApNonce now
Found device in Recovery mode
Identified device as n56ap, iPhone7,1
Got APNonce from device: 0e 53 f3 3e 3f 37 04 17 43 97 e7 2e 43 93 01 fe 94 19 63 c0 
Extracting BuildManifest from IPSW
Product Version: 10.2
Product Build: 14C92 Major: 14
Device supports Image4: true
checking APTicket to be valid for this restore...
Verified ECID in APTicket matches device ECID
checking APTicket to be valid for this restore...
Verified ECID in APTicket matches device ECID
Verified APTicket to be valid for this restore
Variant: Customer Erase Install (IPSW)
This restore will erase your device data.
Extracting filesystem from IPSW
[==================================================] 100.0%
Extracting iBEC.n56.RELEASE.im4p...
Personalizing IMG4 component iBEC...
Sending iBEC (632436 bytes)...
waiting for device to reconnect... 
Getting SepNonce in recovery mode... ef 8b 61 97 2f f4 af e7 4d b0 20 42 b2 95 38 74 ee 8c 84 70 
Getting ApNonce in recovery mode... 0e 53 f3 3e 3f 37 04 17 43 97 e7 2e 43 93 01 fe 94 19 63 c0 
[WARNING] Setting bgcolor to green! If you don't see a green screen, then your device didn't boot iBEC correctly
Recovery Mode Environment:
iBoot build-version=iBoot-3406.30.8
iBoot build-style=RELEASE
Sending RestoreLogo...
Extracting applelogo@3x~iphone.t7000.im4p...
Personalizing IMG4 component RestoreLogo...
Sending RestoreLogo (18834 bytes)...
ramdisk-size=0x10000000
Extracting 058-54560-094.dmg...
Personalizing IMG4 component RestoreRamDisk...
Sending RestoreRamDisk (40330739 bytes)...
Extracting DeviceTree.n56ap.im4p...
Personalizing IMG4 component RestoreDeviceTree...
Sending RestoreDeviceTree (124133 bytes)...
Extracting kernelcache.release.n56...
Personalizing IMG4 component RestoreKernelCache...
Sending RestoreKernelCache (12368816 bytes)...
Trying to fetch new SHSH blob
WARNING: Unable to find BbSkeyId node
Request URL set to https://gs.apple.com/TSS/controller?action=2
Sending TSS request attempt 1... response successfully received
Received SHSH blobs
About to restore device... 
Waiting for device...
Device 2f6bcc7a673907d916f638f3837bccdee8f1aadc is now connected in restore mode...
Connecting now...
Connected to com.apple.mobile.restored, version 14
Device 2f6bcc7a673907d916f638f3837bccdee8f1aadc has successfully entered restore mode
Hardware Information:
BoardID: 4
ChipID: 28672
UniqueChipID: 691215801772070
ProductionMode: true
Starting FDR listener thread
About to send NORData...
Found firmware path Firmware/all_flash/all_flash.n56ap.production
Getting firmware manifest from Firmware/all_flash/all_flash.n56ap.production/manifest
Extracting LLB.n56.RELEASE.im4p...
Personalizing IMG4 component LLB...
Extracting iBoot.n56.RELEASE.im4p...
Personalizing IMG4 component iBoot...
Extracting DeviceTree.n56ap.im4p...
Personalizing IMG4 component DeviceTree...
Extracting applelogo@3x~iphone.t7000.im4p...
Personalizing IMG4 component AppleLogo...
Extracting recoverymode@1920~iphone-lightning.t7000.im4p...
Personalizing IMG4 component RecoveryMode...
Extracting batterylow0@3x~iphone.t7000.im4p...
Personalizing IMG4 component BatteryLow0...
Extracting batterylow1@3x~iphone.t7000.im4p...
Personalizing IMG4 component BatteryLow1...
Extracting batterycharging0@3x~iphone.t7000.im4p...
Personalizing IMG4 component BatteryCharging0...
Extracting batterycharging1@3x~iphone.t7000.im4p...
Personalizing IMG4 component BatteryCharging1...
Extracting glyphplugin@1920~iphone-lightning.t7000.im4p...
Personalizing IMG4 component BatteryPlugin...
Extracting batteryfull@3x~iphone.t7000.im4p...
Personalizing IMG4 component BatteryFull...
Personalizing IMG4 component RestoreSEP...
Personalizing IMG4 component SEP...
Sending NORData now...
Done sending NORData
About to send RootTicket...
Sending RootTicket now...
Done sending RootTicket
Waiting for NAND (28)
Checking filesystems (15)
About to send FDR Trust data...
Sending FDR Trust data now...
Done sending FDR Trust Data
Unmounting filesystems (29)
Unmounting filesystems (29)
Unmounting filesystems (29)
Creating partition map (11)
Creating filesystem (12)
Creating filesystem (12)
Creating filesystem (12)
About to send filesystem...
Connected to ASR
Validating the filesystem
Filesystem validated
Sending filesystem now...
[==================================================] 100.0%
Done sending filesystem
Verifying restore (14)
[==================================================] 100.0%
Mounting filesystems (16)
Mounting filesystems (16)
Mounting filesystems (16)
About to send KernelCache...
Extracting kernelcache.release.n56...
Personalizing IMG4 component KernelCache...
Sending KernelCache now...
Done sending KernelCache
Installing kernelcache (27)
Flashing firmware (18)
[==================================================] 100.0%
Updating gas gauge software (47)
Updating gas gauge software (47)
Updating baseband (19)
About to send BasebandData...
WARNING: Unable to find BbSkeyId node
Sending Baseband TSS request...
Request URL set to https://gs.apple.com/TSS/controller?action=2
TSS server returned: STATUS=126&MESSAGE=An internal error occurred.
ERROR: TSS request failed (status=126, message=An internal error occurred.)
ERROR: Unable to fetch Baseband TSS
ERROR: Unable to send baseband data
ERROR: Unable to successfully restore device
Sending TSS request attempt 1... [Error] ERROR: Unable to restore device
Done: restoring failed.
Failed with errorcode=-11

disappointing...

**Comment by francaisoui** [comment]
@tihmstar does this tool actually work with 6?

**Comment by Shrubbity** [comment]
Try 'irecovery -n' if you encounter this issue. It's what got me out of a boot loop.

**Comment by ibrahimqatar** [comment]
I had the same problem today, I'm so frustrated as it the dev mentioned that if it did not work it wouldn't harm the iphone, yet it forced me to update to 10.2.1

Now I'm trying to downgrade without jailbreak and see what will happen, it's been a hour or so now and nothing matched yet.

**Comment by ibrahimqatar** [comment]
Btw I did this with an iphone 6 plus too. Updated from 9.0.2 to 10.2 and this problem happened

**Comment by xMilesManx** [comment]
Just happened to me too. RIP jailbreak

**Comment by tihmstar** [comment]
same as #110

**Comment by dirtabd** [comment]
May have made the same mistake i did, if a nonce was set i believe there should be no -w before your ipsw file, I made the mistake of typing it out myself and for some reason at that last command I looked at the b option and added the -w before my ipsw. I believe the -w is to try and match the nonce code by throwing every possibility at it constantly? I'm still new to this, is that in the right area for the more serious coders out there? can anyone confirm whether that would output the error 11 or did Futurestore just fail somehow?
